tests/test_manager.py

```diff
@@ -595,6 +595,7 @@ async def test_unknown_user(
         form = create_oauth2_password_request_form("lancelot@camelot.bt", "guinevere")
         user = await user_manager.authenticate(form)
         assert user is None
+        assert user_manager.on_after_login.called is False
```

Comment:
The test assertions should be in [`TestLogin`](https://github.com/fastapi-users/fastapi-users/blob/a665cd5ed731679b342ec9c2869175447358ddba/tests/test_router_auth.py#L59) and [`TestCallback`](https://github.com/fastapi-users/fastapi-users/blob/a665cd5ed731679b342ec9c2869175447358ddba/tests/test_router_oauth.py#L141): since we call the hook in the router, it won't work here since we are only testing the `authenticate` method.